File: 프로그래머스/LV1/코딩테스트 연습/연습문제/가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(string s) 
{
    map<char,vector<int>> alphabet_idxs_map;
    vector<int> v;
    vector<int> answer;

    for(int i = 0; i < s.length(); i++)
    {
        if(alphabet_idxs_map.find(s[i]) == alphabet_idxs_map.end())
        {
            v = {i};
            alphabet_idxs_map.insert(make_pair(s[i], v));
            answer.push_back(-1);
        }
        else
        {
            answer.push_back(i - alphabet_idxs_map[s[i]].back());
            alphabet_idxs_map[s[i]].push_back(i);
        }
    }
    return answer;
}

File: 프로그래머스/LV1/코딩테스트 연습/연습문제/가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    alphabet_idxs_dict = dict()
    answer = []
    for i in range(len(s)):
        if s[i] not in alphabet_idxs_dict:
            alphabet_idxs_dict[s[i]] = [i]
            answer.append(-1)
        else:
            answer.append(i - alphabet_idxs_dict[s[i]][-1])
            alphabet_idxs_dict[s[i]].append(i)
    return answer